bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/target/env/TargetEnvironment.java

```diff
@@ -89,10 +91,25 @@ public static TargetEnvironment create(SPTarget base, GuideEnvironment guide, Im
     }
 
     /** Returns an arbitrary target from the asterism. This method will be removed for 18A. */
+    public SPSkyObject getPrimaryTargetFromAsterism() {
+        if (asterism instanceof GhostAsterism) {
+            final GhostAsterism ga = (GhostAsterism) asterism;
+            if (ga.base().isDefined())
+                return ga.base().get();
+            else if (ga instanceof GhostAsterism.DualTarget) {
+                final Option<SPCoordinates> c =
+                        ImOption.fromScalaOpt(ga.basePosition(ImOption.scalaNone())).map(SPCoordinates::new);
```

Comment:
Looking at it here, without doubling back to check the definition, it isn't clear to me what the difference is between `base` and `basePosition`.

Reply:
Maybe `getOverriddenBase` for the `Option[SPCoordinates]` one?

Reply:
Do you mean changing this in `GhostAsterism`? I'm a bit confused.

Reply:
Yes, just a renaming in `GhostAsterism` that makes the difference clearer.  One is an explicitly defined "override" base position and may not be set at all.  The other is an algorithm that gives you a base position regardless.  So maybe `overriddenBase` vs `calculatedBase`?  I dunno.  Good names don't occur to me.  `getBase` vs `getBasePosition` doesn't clear to me though.